src/imageops/mod.rs

```diff
@@ -61,6 +63,11 @@ pub fn simple_decode (img: &RawImage, maxwidth: usize, maxheight: usize) -> OpBu
   // Demosaic into 4 channel f32 (RGB or RGBE)
   let mut channel4 = do_timing("demosaic", ||demosaic::demosaic_and_scale(img, maxwidth, maxheight));
 
+  // Fix orientation if necessary and possible
+  if img.orientation != Orientation::Normal && img.orientation != Orientation::Unknown {
+    channel4 = do_timing("rotate", || { transform::rotate(img, &channel4) });
```

Comment:
When there is a transpose this will break the maxwidth/maxheight contract. We'll need proper pipeline sizing code at some point but for now just detecting those cases and flipping maxwidth/maxheight is enough.